`0.2.120` causes compilation failure `error[E0658]: use of unstable library feature simd_wasm64`

With today's release of `0.2.120` I have started seeing [compilation failures in CI](https://github.com/smithy-lang/smithy-rs/actions/runs/25077367361/job/73476548782?pr=4628) caused by `wasm-bindgen` and the use of the `simd_wasm64` feature. Haven't had a chance to dive into it fully yet, but the linked PR (and the other PRs it has shown up on) don't change anything about our WASM use besides bumping the version of `wasm-bindgen` in `Cargo.lock` from `0.2.118` to `0.2.120` ([example change](https://github.com/smithy-lang/smithy-rs/pull/4628/changes#diff-48581665017cc79e36421593df21ccd7222563c6e7e7b2dfd524c69af3ddce6dL2481))

I'll look at the wasm-bindgen release diff in a bit, but this seems like a breaking change to me. Note that we are on rust/cargo version `1.91.1` so if this stabilized recently we might just need to do an MSRV bump to catch up.

Full error below:

```
error[E0658]: use of unstable library feature `simd_wasm64`
   --> /opt/cargo/registry/src/index.crates.io-1949cf8c6b5b557f/wasm-bindgen-0.2.120/src/externref.rs:112:13
    |
112 |             core::arch::wasm::unreachable();
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #90599 <https://github.com/rust-lang/rust/issues/90599> for more information

error[E0658]: use of unstable library feature `simd_wasm64`
   --> /opt/cargo/registry/src/index.crates.io-1949cf8c6b5b557f/wasm-bindgen-0.2.120/src/rt/mod.rs:729:13
    |
729 |             core::arch::wasm::unreachable();
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #90599 <https://github.com/rust-lang/rust/issues/90599> for more information

For more information about this error, try `rustc --explain E0658`.
error: could not compile `wasm-bindgen` (lib) due to 2 previous errors
warning: build failed, waiting for other jobs to finish...

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,19 @@
 # `wasm-bindgen` Change Log
 --------------------------------------------------------------------------------
 
+## Unreleased
+
+### Added
+
+### Fixed
+
+* Fix compilation failure with `no_std` + `release`
+  [#5134](https://github.com/wasm-bindgen/wasm-bindgen/pull/5134)
+
+### Changed
+
+--------------------------------------------------------------------------------
+
 ## [0.2.119](https://github.com/rustwasm/wasm-bindgen/compare/0.2.118...0.2.119)
 
 ### Added

--- a/src/externref.rs
+++ b/src/externref.rs
@@ -108,8 +108,12 @@ fn internal_error(_msg: &str) -> ! {
             super::throw_str(_msg)
         } else if #[cfg(feature = "std")] {
             std::process::abort();
-        } else if #[cfg(target_family = "wasm")] {
-            core::arch::wasm::unreachable();
+        } else if #[cfg(target_arch = "wasm32")] {
+            // stable
+            core::arch::wasm32::unreachable();
+        } else if #[cfg(target_arch = "wasm64")] {
+            // unstable, need simd_wasm64 feature
+            core::arch::wasm64::unreachable();
         } else {
             unreachable!()
         }

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -46,6 +46,10 @@
     feature(allow_internal_unstable),
     allow(internal_features)
 )]
+#![cfg_attr(
+    all(not(debug_assertions), not(feature = "std"), target_arch = "wasm64"),
+    feature(simd_wasm64)
+)]
 #![doc(html_root_url = "https://docs.rs/wasm-bindgen/0.2")]
 
 extern crate alloc;

--- a/src/rt/mod.rs
+++ b/src/rt/mod.rs
@@ -722,11 +722,12 @@ fn malloc_failure() -> ! {
             super::throw_str("invalid malloc request")
         } else if #[cfg(feature = "std")] {
             std::process::abort();
-        } else if #[cfg(all(
-            target_family = "wasm",
-            any(target_os = "unknown", target_os = "none")
-        ))] {
-            core::arch::wasm::unreachable();
+        } else if #[cfg(target_arch = "wasm32")] {
+            // stable
+            core::arch::wasm32::unreachable();
+        } else if #[cfg(target_arch = "wasm64")] {
+            // unstable, need simd_wasm64 feature
+            core::arch::wasm64::unreachable();
         } else {
             unreachable!()
         }

